### ISSUE TITLE
Global error handler detects errors only once

### ISSUE THREAD

**Issue by wachri** [angular/angular#14949]
**I'm submitting a ...** 
```
[x ] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
The global error handler logs and detects error only once. After that all other errors do not get logged to the console.

**Expected behavior**
Every error get's logged to the console regardless of whether a error was thrown before or not.

**Minimal reproduction of the problem with instructions**
1. Go to https://plnkr.co/edit/EPXTqlN0hvJGavNPVfm7?p=preview
1. Open console
1. Click a button
1. Click a second time on a button

* **Angular version:** 2.4.9

* **Browser:** [all]

 
* **Language:** [TypeScript 2.X]

**Comment by DzmitryShylovich** [comment]
Duplicate of https://github.com/angular/angular/issues/13159

**Comment by DzmitryShylovich** [comment]
@pkozlowski-opensource duplicate

**Comment by chuckjaz** [comment]
Change was reverted.

**Comment by DzmitryShylovich** [comment]
@chuckjaz why? it doesn't fail bootstrap on error?

**Comment by chuckjaz** [comment]
We are not comfortable with the scope of the change given how close we are to final of 4.0.0. We will reconsider the change for 4.1 when we have more time to evaluate the impact.

**Comment by DzmitryShylovich** [comment]
Workaround - https://plnkr.co/edit/CJ2zP7gPJnSTDjCeWviI?p=preview

**Comment by mhevery** [comment]
ErrorHandler can not throw errors because it will unsubscribe itself from
the error stream.

Zones captures errors and feed it into NgZone, which than has a Rx Observable
to feed it into ErrorHandler. If the ErroHandler throws, then Rx will teardown
the observable which in essence causes the ErrorHandler to be removed from the
error handling. This implies that the ErrorHandler can never throw errors.

**Comment by 4231** [comment]
Globally subscribe defined or not when return HTTP request data
